lib/msal-node/test/client/ManagedIdentityApplication.spec.ts

```diff
@@ -0,0 +1,326 @@
+                const managedIdentityApplication =
+                    new ManagedIdentityApplication(systemAssignedConfig);
+                expect(managedIdentityApplication.isIMDS()).toBe(true);
+
+                const networkManagedIdentityResult =
+                    await managedIdentityApplication.acquireToken({
+                        resource: MANAGED_IDENTITY_RESOURCE,
+                    });
+                expect(networkManagedIdentityResult.fromCache).toBe(false);
+
+                expect(networkManagedIdentityResult.accessToken).toEqual(
+                    DEFAULT_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+                );
+                expect(
+                    managedIdentityApplication.getConfig.managedIdentityId.getId
+                ).toEqual(DEFAULT_MANAGED_IDENTITY_ID);
+                expect(
+                    managedIdentityApplication.getConfig.managedIdentityId
+                        .getIdType
+                ).toEqual(ManagedIdentityIdType.SYSTEM_ASSIGNED);
+                expect(
+                    managedIdentityApplication.getConfig.managedIdentityId
+                        .getIsUserAssignedId
+                ).toEqual(false);
+
+                // wait two seconds for the token to be saved to the cache, before checking the cache for the token
+                // await new Promise<void>((resolve) => {
+                //     let counter: number = 0;
+                //     const interval = setInterval(() => {
+                //         if (counter === 2000) {
+                //             resolve();
+                //             clearInterval(interval);
+                //         } else {
+                //             counter++;
+                //         }
+                //     }, 1); // 1 millisecond
+                // });
+
+                const cacheManagedIdentityResult =
+                    await managedIdentityApplication.acquireToken({
+                        resource: MANAGED_IDENTITY_RESOURCE,
+                    });
+                expect(cacheManagedIdentityResult.fromCache).toBe(true);
+                expect(cacheManagedIdentityResult.accessToken).toEqual(
+                    DEFAULT_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+                );
+                expect(
+                    managedIdentityApplication.getConfig.managedIdentityId.getId
+                ).toEqual(DEFAULT_MANAGED_IDENTITY_ID);
+                expect(
+                    managedIdentityApplication.getConfig.managedIdentityId
+                        .getIdType
+                ).toEqual(ManagedIdentityIdType.SYSTEM_ASSIGNED);
+                expect(
+                    managedIdentityApplication.getConfig.managedIdentityId
+                        .getIsUserAssignedId
+                ).toEqual(false);
+            });
+        });
+    });
+});
```

Comment:
We need tests that validate the access token cache key for all configurations - system assigned, user assigned + client etc.

Reply:
Updated the api review doc to add cache keys for all the managed identity scenarios. https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=/ManagedIdentity/ManagedIdentity.md&version=GBdev&_a=preview&anchor=acceptance-tests-for-cache-keys.

Reply:
I just read Neha's addition to the API review doc regarding the cache keys. I don't think I can write unit tests to verify them... the nodeStorage variable is private and can't be accessed from the unit tests. In Client Credential, we can pass in a custom node storage, and verify cache items that way. We've disabled that for Managed Identity, so I don't think I will be able to have access to the cache to check the cache key.